Remove unused imports from docs tests

Harsh1s opened the pull request:
<!--
Thank you for your pull request to ICU4X!

Reminder: try to use [Conventional Comments](https://conventionalcomments.org/) to make comments clearer.

Please see https://github.com/unicode-org/icu4x/blob/main/CONTRIBUTING.md for general
information on contributing to ICU4X.
-->
##Description:
Cleaned all the unused imports, which can be confusing to clients and developers.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/unicode-org/icu4x?pullRequest=4628) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/unicode-org/icu4x?pullRequest=4628) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/unicode-org/icu4x?pullRequest=4628) it.</sub>

sffc commented:
Thanks for doing this! Two things:

1. Please check the `test-docs` CI and make sure all the docs tests are still passing.
2. As @robertbastian suggested, please modify the job to run with the correct flags to prevent backsliding. You can set the flags in `[tasks.test-docs]` in `tools/make/tests.toml` here: https://github.com/unicode-org/icu4x/blob/4242dfbdc7938d036565374e973e37d97b911f41/tools/make/tests.toml#L49

sffc reviewed (changes requested):
^

Harsh1s commented:
I have fixed the failing test now, it happened cuz I removed a legit import accidently.

Now about the `test.toml` arguements, from my research, I found out that `rustdoc` does not currently warn about unused imports. It’s a known limitation of `rustdoc`.  So since there are no warnings generated, it is kind of difficult to prevent backsliding as you guys adviced. I'm still trying a few things but just wanted to keep you in the loop.

Harsh1s commented:
Here's an update of what I found:
"You can show warnings in doctests by running `rustdoc --test --test-args=--show-output`" is what the rust documentation says. So I went ahead and ran `RUSTFLAGS="-D warnings" RUSTFLAGS="--test-args --show-output" cargo test --all-features --doc --no-fail-fast` on the command line. Unfortunately, there was not a single warning concerning unused imports. This makes it kind of complicated as no warnings mean nothing to abort or fix automatically.

@robertbastian I hope I'm not doing something absolutely stupid, would like to hear your advice on this.

robertbastian reviewed (approved):
I've looked into this and couldn't find a way to do this either. There's an open issue for this (https://github.com/rust-lang/rust/issues/56232), but no workaround afaict. Let's merge the manual fixes for now.

Edit: please fix broken tests

Harsh1s commented:
> I've looked into this and couldn't find a way to do this either. There's an open issue for this ([rust-lang/rust#56232](https://github.com/rust-lang/rust/issues/56232)), but no workaround afaict. Let's merge the manual fixes for now.
> 
> Edit: please fix broken tests

Yeah unfortunately it was a wild goose chase :( 
On the bright side though, manually removing them helped me understand the project more. Hope to make good contributions moving forward!

P.S. I've fixed the broken tests, @robertbastian thanks for your help with the issue!
